pychromecast/__init__.py

```diff
@@ -334,6 +344,33 @@ def join(self, timeout=None):
         """
         self.socket_client.join(timeout=timeout)
 
+    def start_dynamic_host_tracking(self):
+        """
+        Start dynamic non-blocking host tracking
+        """
+        self.tracker, self.browser = \
+            start_discovery(self._dynamic_host_callback)
+
+    def _dynamic_host_callback(self, unit_id):
+        """
+        Callback for dynamic discovery of host ip changes
+        """
+        unit = self.tracker.services[unit_id]
+        host = unit[0]
+        port = unit[1]
+        name = unit[4]
+
+        if not name == self.name or not port == self.port:
+            return  # Wrong group, keep discovering
+
+        if host == self.host:
+            return  # No change in host, keep discovering
+
+        self.logger.info('Change in host ip detected, '
+                         'reinitializing object...')
+        stop_discovery(self.browser)
+        self.__init__(host, port=port, device=self.device, dynamic_host=True)
```

Comment:
Yeah, that's pretty much it. AFAIK there isn't a technical reason it can't be called twice, it's just a confusing pattern.

For cleanup, you'll probably just need to call `self.disconnect`. You'll want to keep the default of blocking=True to ensure it closes successfully first.